src/solver.c

```diff
@@ -371,119 +371,198 @@ void transport_sweep( Params * params, Input * I )
+                                    printf("Error: invalid axial expansion order");
+                                    printf("\n Please input 0 or 2\n");
+                                    exit(1);
+                                }
+                            }
+                        }
+
+                        // determine if there are tracks that cross the entire
+                        // z-height of the source region (mutually exclusive
+                        // with any traveling the entire 2D distance)
+                        else if( start_full > end_full )
+                        {
+                            float ds = (z_max - z_min) / abs(mu);
+
+                            for (int k = end_full; k < start_full; k++) {
+
+                                // select current track
+                                Track * track = &params->tracks[i][j][k];
+                                
+                                /* update sources and fluxes from attenuation 
+                                 * over SR */
+                                if( I->axial_exp == 2 )
+                                {
+                                    attenuate_fluxes( track, true, 
+                                            &params->sources[QSR_id], 
+                                            I, params, ds, mu, 
+                                            params->tracks_2D[i].az_weight, &A );
+
+                                    segments_processed++;
+                                }
+                                else if( I->axial_exp == 0 )
+                                {
+                                    attenuate_FSR_fluxes( track, true,
+                                            &params->sources[QSR_id],
+                                            I, params, ds, mu,
+                                            params->tracks_2D[i].az_weight, &A );
+
+                                    segments_processed++;
+                                }
+                                else
+                                {
+                                    printf("Error: invalid axial expansion order");
+                                    printf("\n Please input 0 or 2\n");
+                                    exit(1);
+                                }
+                            }
+                        }
+                        // Treat upper tracks that do not cross the entire 2D
+                        // length
+                        int min_upper = max(start_full, end_full);
+                        for( int k = min_upper; k < end_track; k++) {
+
+                            // select current track
+						    Track * track = &params->tracks[i][j][k];
+                            
+                            // calculate the distance traveled in the SR
+                            float start_z = first_track_lower_z + k*fine_delta_z;
+                            float ds = (z_max - start_z) / abs(mu);
```

Comment:
Is the integer abs() call intentional here? If you want the floating point one, that would be fabs or fabsf.